sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/functions/ScalarFunction.java

```diff
@@ -117,6 +120,8 @@
  *   <li>{@link org.apache.spark.sql.types.DateType}: {@code int}</li>
  *   <li>{@link org.apache.spark.sql.types.TimestampType}: {@code long}</li>
  *   <li>{@link org.apache.spark.sql.types.BinaryType}: {@code byte[]}</li>
+ *   <li>{@link org.apache.spark.sql.types.CalendarIntervalType}:
```

Comment:
this is not related - lemme know if I should use a separate PR.

Reply:
I'm fine. :)

Reply:
I'm wondering if we should forbid the old interval type in v2 function as it's a new feature. cc @MaxGekk

Reply:
ohh I'm happy to remove it then if there is any concern. We can add it separately later.